pkgs/development/libraries/qt-6/default.nix

```diff
@@ -0,0 +1,124 @@
+        inherit (srcs.qtbase) src version;
+        inherit bison cups harfbuzz libGL dconf gtk3 developerBuild decryptSslTraffic cmake;
+        inherit (darwin.apple_sdk.frameworks) AGL AppKit ApplicationServices Carbon Cocoa CoreAudio CoreBluetooth
+          CoreLocation CoreServices DiskArbitration Foundation OpenGL MetalKit IOKit;
+        inherit (darwin) libobjc;
+      };
+
+      qt3d = callPackage ./modules/qt3d.nix { };
+      qt5compat = callPackage ./modules/qt5compat.nix { };
+      qtcharts = callPackage ./modules/qtcharts.nix { };
+      qtconnectivity = callPackage ./modules/qtconnectivity.nix { };
+      qtdatavis3d = callPackage ./modules/qtdatavis3d.nix { };
+      qtdeclarative = callPackage ./modules/qtdeclarative.nix { };
+      qtdoc = callPackage ./modules/qtdoc.nix { };
+      qtimageformats = callPackage ./modules/qtimageformats.nix { };
+      qtlanguageserver = callPackage ./modules/qtlanguageserver.nix { };
+      qtlottie = callPackage ./modules/qtlottie.nix { };
+      qtmultimedia = callPackage ./modules/qtmultimedia.nix {
+        inherit gstreamer gst-plugins-base;
+      };
+      qtnetworkauth = callPackage ./modules/qtnetworkauth.nix { };
+      qtpositioning = callPackage ./modules/qtpositioning.nix { };
+      qtsensors = callPackage ./modules/qtsensors.nix { };
+      qtserialbus = callPackage ./modules/qtserialbus.nix { };
+      qtserialport = callPackage ./modules/qtserialport.nix { };
+      qtshadertools = callPackage ./modules/qtshadertools.nix { };
+      qtquick3d = callPackage ./modules/qtquick3d.nix { };
+      qtquicktimeline = callPackage ./modules/qtquicktimeline.nix { };
+      qtremoteobjects = callPackage ./modules/qtremoteobjects.nix { };
+      qtsvg = callPackage ./modules/qtsvg.nix { };
+      qtscxml = callPackage ./modules/qtscxml.nix { };
+      qttools = callPackage ./modules/qttools.nix { };
+      qttranslations = callPackage ./modules/qttranslations.nix { };
+      qtvirtualkeyboard = callPackage ./modules/qtvirtualkeyboard.nix { };
+      qtwayland = callPackage ./modules/qtwayland.nix { };
+      qtwebchannel = callPackage ./modules/qtwebchannel.nix { };
+      qtwebengine = callPackage ./modules/qtwebengine.nix {
+        inherit (darwin) cctools libobjc libunwind xnu;
+        inherit (darwin.apple_sdk.libs) sandbox;
+        inherit (darwin.apple_sdk.frameworks) ApplicationServices AVFoundation Foundation ForceFeedback GameController AppKit
+          ImageCaptureCore CoreBluetooth IOBluetooth CoreWLAN Quartz Cocoa LocalAuthentication;
+      };
+      qtwebsockets = callPackage ./modules/qtwebsockets.nix { };
+      qtwebview = callPackage ./modules/qtwebview.nix {
+        inherit (darwin.apple_sdk.frameworks) CoreFoundation WebKit;
+      };
+
+      wrapQtAppsHook = makeSetupHook
+        {
+          deps = [ self.qtbase.dev makeWrapper ]
+            ++ lib.optional stdenv.isLinux self.qtwayland.dev;
+        } ../qt-5/hooks/wrap-qt-apps-hook.sh;
```

Comment:
Does a single setup hook worth it's own directory? There's not much code to share between qt 5 and 6 beyond that.

Reply:
just copy to `qt-6/hooks/wrap-qt-apps-hook.sh`?
see also qt-5/README.md

> ## Major Updates
> 
> Let `$major` be the new major version number, e.g. `5.10`.
> 
> 1. Copy the subdirectory from the previous major version to `$major`.

Reply:
But we have no intention to support multiple versions of qt6 as previously discussed, and have a bunch of duplicated files in nixpkgs is not the best thing I could imagine.

Reply:
> we have no intention to support multiple versions of qt6

multiple \*minor\* versions

> have a bunch of duplicated files in nixpkgs

this is one file with 2612 bytes. nixpkgs has 260 MB, so that's 0.001 %
also, if it's exactly the same file, it's deduped by git